libs/blog/shared/feature-seo/src/lib/seo/seo.service.ts

```diff
@@ -0,0 +1,357 @@
+      this.setMetaTwitterMisc(seoData.twitter_misc);
+    }
+  }
+
+  private setMetaTwitterMisc(miscData: object): void {
+    const entries = Object.entries(miscData);
+
+    for (const [index, entry] of entries.entries()) {
+      const label: MetaDefinition = {
+        name: `${SEO_META_KEYS.twitterMiscLabel}${index + 1}`,
+        content: entry[0],
+      };
+
+      const data: MetaDefinition = {
+        name: `${SEO_META_KEYS.twitterMiscData}${index + 1}`,
+        content: entry[1],
+      };
+
+      this._meta.updateTag(label);
+      this._meta.updateTag(data);
+    }
+  }
+
+  private setMetaTwitterCard(content: string): void {
+    const metaTwitterCard: MetaDefinition = {
+      name: SEO_META_KEYS.twitterCard,
+      content: content,
+    };
+
+    this._meta.updateTag(metaTwitterCard);
+  }
+
+  private setMetaModifiedTime(modifiedTime: string): void {
+    const metaModifiedTime: MetaDefinition = {
+      name: SEO_META_KEYS.articleModifiedTime,
+      content: modifiedTime,
+    };
+
+    this._meta.updateTag(metaModifiedTime);
+  }
+
+  private setMetaPublishedTime(publishedTime: string): void {
+    const metaPublishedTime: MetaDefinition = {
+      name: SEO_META_KEYS.articlePublishedTime,
+      content: publishedTime,
+    };
+
+    this._meta.updateTag(metaPublishedTime);
+  }
+
+  private setMetaPublisher(publisher: string): void {
+    const metaPublisher: MetaDefinition = {
+      name: SEO_META_KEYS.articlePublisher,
+      content: publisher,
+    };
+
+    this._meta.updateTag(metaPublisher);
+  }
+
+  private setMetaRobots(robots: SeoRobotsData): void {
+    const content = Object.values(robots).join(', ');
+
+    const metaRobots: MetaDefinition = {
+      name: SEO_META_KEYS.robots,
+      content: content,
+    };
+
+    this._meta.updateTag(metaRobots);
+  }
+
+  private setMetaLocale(locale: string): void {
+    const metaLocale: MetaDefinition = {
+      name: SEO_META_KEYS.ogLocale,
+      content: locale,
+    };
+
+    this._meta.updateTag(metaLocale);
+  }
+
+  private setMetaDescription(metaDescription: string): void {
+    const description: MetaDefinition = {
+      name: SEO_META_KEYS.description,
+      content: metaDescription,
+    };
+    const ogDescription: MetaDefinition = {
+      property: SEO_META_KEYS.ogDescription,
+      content: metaDescription,
+    };
+    const twitterDescription: MetaDefinition = {
+      name: SEO_META_KEYS.twitterDescription,
+      content: metaDescription,
+    };
+    const googleDescription: MetaDefinition = {
+      itemprop: SEO_META_KEYS.description,
+      content: metaDescription,
+    };
+
+    this._meta.removeTag('itemprop="description"'); // Because if we not remove the tag it will not be updated.
+
+    this._meta.updateTag(description);
+    this._meta.updateTag(ogDescription);
+    this._meta.updateTag(twitterDescription);
+    this._meta.updateTag(googleDescription);
+  }
+
+  private setMetaType(metaType: string): void {
```

Comment:
i would reduce number of code lines with that func: 

```
 private updateTag(content: string, name: SeoMetaKeys): void {
    const meta: MetaDefinition = {
      name: name,
      content: content,
    };

    this._meta.updateTag(meta);
  }
```

where SeoMetaKeys

`export type SeoMetaKeys = keyof typeof SEO_META_KEYS;`

and 

`SEO_META_KEYS = {...} as const`

Reply:
Done :)